pq/autocomplete/earley/earley.go

```diff
@@ -121,28 +121,34 @@ func (p *Earley) resizeChartIfNecessary(words Tokens) {
 }
 
 func (p *Earley) resizeChart(size int) {
+	p.chart.resetChart()
```

Comment:
We should probably only reset under certain conditions (it does make sense in the test, however). Upsizing the chart size is helpful when our input string is growing during runtime, since we can incrementally compute our chart.

Reply:
That's reason why there is a ParseTokens and also PartialParse. I finally got it. 
Then I think ParseToken is to parse the a new input, we'd better clean up the old stateset because the leftover will be replaced anyway. If the previous stateset is longer than current, it will cause unwanted confusion.

For PartialParse, we need to keep stateset and incrementally parse.
WDYT?